standard/variables.md

```diff
@@ -1012,50 +1012,50 @@ A *reference variable* is a variable that refers to another variable, called the
+>         ref int element = ref arr[5];
+>         return ref element; // return reference to arr[5];
+>     }     
 > }
 > ```
 >
 > *end example*
 
-A *reference return* is the expression returned by reference from a method whose return type includes the `ref` or `ref readonly` modifiers (§14.6.1). The variable expression of the *reference return* is the *referent* of the *reference return*.
+A `ref struct` may include `ref struct` or ref-like fields. A *ref-like field* refers to the same storage as its initializing expression. The initializing expression of the *ref_like field* is its *referent*. Unlike a reference field, a ref-like field may refer to a `struct` whose storage may be on the execution stack. Ref struct types include `Span<T>`, `ReadOnlySpan<T>`, and other types that include an unmanaged pointer as a member.
 
-> *Example:* The following example demonstrates a =*reference return* whose *referent* is an element of an array field:
+> *Example:* The following example demonstrates a `ReadOnlySpan<T>` *reference variable* whose *referent* is the characters in a string. `ReadOnlySpan<T>` is a `ref struct`. One of its internal fields is a *ref-like field* that refers to the character array in the `string` referent. Therefore, `chars` is a *reference variable*:
 >
 > ```csharp
-> public class C
+> string Reverse(string s)
 > {
->     private int[] arr = new int[10];
->
->     public readonly ref int M()
+>     ReadOnlySpan<char> chars = s.AsSpan();
+>     var reversed = new char[chars.Length];
+>     for (int i = 0; i < chars.Length; i++)
 >     {
->         // element is a reference variable that refers to arr[5]
->         ref int element = ref arr[5];
->         return ref element; // return reference to arr[5];
->     }     
+>         reversed[chars.Length-i-1] = chars[i];
+>     }
+>     return new string(reversed);
 > }
 > ```
 >
 > *end example*
 
 ### §ref-span-safety-escape-scopes Ref safe scopes
 
-All reference variables obey safety rules that ensure the scope of the reference variable is not greater than the *ref safe scope* of its referent.
+All reference variables obey safety rules that ensure the scope of the reference variable is not greater than the *ref_safe_scope* of its referent.
 
-For any variable, the *ref_safe_scope* of that variable is the scope where a *variable_reference* (§9.5) to that variable is valid. The *referent* of a reference variable must have a *ref_safe_scope* that is at least as wide as the scope where the reference variable is valid.
+For any variable, the *ref_safe_scope* of that variable is the scope where a *variable_reference* (§9.5) to that variable is valid. The *referent* of a reference variable must have a *ref_safe_scope* that is at least as wide as the scope of the reference variable. A `ref struct` instance may not be copied by value beyond the value *ref_safe_scope* of the initializing expression(s) of any *ref-like fields*.
```

Comment:
Here's the additional sentence.

Reply:
Removed from this PR, and added to the appropriate location on #601 

After thinking about Neal's points yesterday, I think the clearest way to spec this will be to have the ref safety rules in Variables. Those rules relating to the safe scope for a ref struct should be in the clause on ref structs. The rules apply, but trivially, to variables of other types. (Namely, ref structs are constrained to a safe scope. So are other variables, but for non-ref structs, the safe scope is calling method. Only ref structs have "interesting" constraints on where they can be copied.